include/nbl/ui/IWindowManager.h

```diff
@@ -1,11 +1,14 @@
 #ifndef I_WINDOWMANAGER
 #define I_WINDOWMANAGER
-
+#include <nbl/core/IReferenceCounted.h>
+#include "IWindow.h"
 namespace nbl::ui
 {
-	class IWindowManager
+	class IWindowManager : core::IReferenceCounted
```

Comment:
public inheritance

Reply:
public virtual inheritance

Reply:
why virtual

Reply:
because almost everything inherits from `IReferenceCounted` so you eventually always run into diamonds involving `IReferenceCounted`, plus you really dont want two copies of IReferenceCounted in your hierarchy (think about how having two ref counters would fuck everything up in terms of autodestruction)

Reply:
yea, i just tend to avoid virtual inheritance unless theres an actual need

Reply:
yeah IReferenceCounted is the only place you want to be using virtual inheritance